src/effects/auth/index.js

```diff
@@ -80,3 +80,28 @@ export function useRedirectIfLoggedIn(location = "/"): void {
     );
   });
 }
+
+export function useRedirectIfNoControlAccessToPod(
+  location = "/access-required"
+) {
+  const { session, isLoadingSession } = useContext(SessionContext);
+  const profile = useAuthenticatedProfile();
+
+  useEffect(() => {
+    if (isLoadingSession || !profile) return;
+
+    Promise.all(
+      profile.pods.map((pod) =>
```

Comment:
Rather than reading all pods for a profile, we should look at the current pod (especially in the case that we're visiting someone else's pod.) Use `PodLocationContext` for the current iri, and get its dataset.

Reply:
But we should be able to visit other Pods without having Control access, no?

I agree that we should have some access check for pods you're visiting, but I thought this PR was suppose to check that you have Control access to the Pods you have listed in your profile?

Reply:
Even if that's true, we shouldn't be reading the pod iri from the profile. This should be checking the current pod.

That said, we should only run `useRedirectIfNoControlAccessToPod` if you're on your own pod, which would require checking the current pod iri against the list from your profile.

Reply:
Ok, so I'll change the logic to the following:

* `useRedirectIfNoControlAccessToOwnPod`: This will kick in _if_ you're visiting your own Pod (i.e. one of the ones listed in your profile) _and_ you don't have Control access
* `useRedirectIfNoAccessToPod`: This will kick in _if_ you're trying to visit any Pod that you don't have Read access to

Does that sound reasonable?

Reply:
Yes, that sounds reasonable.

Reply:
Think it does what you expect now.